src/framework/ui/uiwidget.cpp

```diff
@@ -247,6 +250,10 @@ void UIWidget::insertChild(size_t index, const UIWidgetPtr& child)
     updateChildrenIndexStates();
 
     g_ui.onWidgetAppear(child);
+    callLuaField("onInsertChild", child);
+
+    if (getRootParent() == g_ui.getRootWidget())
+        child->onAdopted(static_self_cast<UIWidget>());
```

Comment:
This is going to be a spam call to lua in certain cases, I don't see much use case for this to be hones, but if there is a good reason, then it should be disabled by default like we do in tile.cpp for example:

```
    if (g_game.isTileThingLuaCallbackEnabled())
        callLuaField("onAddThing", thing);
 ```